runtime/src/gpu/cpu/gpu-cpu.c

```diff
@@ -201,4 +201,61 @@ void* chpl_gpu_impl_host_register(void* var, size_t size) { return var; }
 void chpl_gpu_impl_host_unregister(void* var) { }
 
 #undef DEF_ONE_SORT
+
+void chpl_gpu_impl_name(int dev, char **result) {
+  char* name = (char *)chpl_mem_alloc(0xFF, CHPL_RT_MD_IO_BUFFER, __LINE__, 0);
+  strcpy(name, "chapel-cpu-as-device-gpu");
+  *result = name;
+}
+
+const int CHPL_GPU_ATTRIBUTE__MAX_THREADS_PER_BLOCK = 0;
+const int CHPL_GPU_ATTRIBUTE__MAX_BLOCK_DIM_X = 1;
+const int CHPL_GPU_ATTRIBUTE__MAX_BLOCK_DIM_Y = 2;
+const int CHPL_GPU_ATTRIBUTE__MAX_BLOCK_DIM_Z = 3;
+const int CHPL_GPU_ATTRIBUTE__MAX_GRID_DIM_X = 4;
+const int CHPL_GPU_ATTRIBUTE__MAX_GRID_DIM_Y = 5;
+const int CHPL_GPU_ATTRIBUTE__MAX_GRID_DIM_Z = 6;
+const int CHPL_GPU_ATTRIBUTE__MAX_SHARED_MEMORY_PER_BLOCK = 7;
+const int CHPL_GPU_ATTRIBUTE__TOTAL_CONSTANT_MEMORY = 8;
+const int CHPL_GPU_ATTRIBUTE__WARP_SIZE = 9;
+const int CHPL_GPU_ATTRIBUTE__MAX_PITCH = 10;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE1D_WIDTH = 11;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE2D_WIDTH = 12;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE2D_HEIGHT = 13;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE3D_WIDTH = 14;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE3D_HEIGHT = 15;
+const int CHPL_GPU_ATTRIBUTE__MAXIMUM_TEXTURE3D_DEPTH = 16;
+const int CHPL_GPU_ATTRIBUTE__MAX_REGISTERS_PER_BLOCK = 17;
+const int CHPL_GPU_ATTRIBUTE__CLOCK_RATE = 18;
+const int CHPL_GPU_ATTRIBUTE__TEXTURE_ALIGNMENT = 19;
+const int CHPL_GPU_ATTRIBUTE__TEXTURE_PITCH_ALIGNMENT = 20;
+const int CHPL_GPU_ATTRIBUTE__MULTIPROCESSOR_COUNT = 21;
+const int CHPL_GPU_ATTRIBUTE__KERNEL_EXEC_TIMEOUT = 22;
+const int CHPL_GPU_ATTRIBUTE__INTEGRATED = 23;
+const int CHPL_GPU_ATTRIBUTE__CAN_MAP_HOST_MEMORY = 24;
+const int CHPL_GPU_ATTRIBUTE__COMPUTE_MODE = 25;
+const int CHPL_GPU_ATTRIBUTE__CONCURRENT_KERNELS = 26;
+const int CHPL_GPU_ATTRIBUTE__ECC_ENABLED = 27;
+const int CHPL_GPU_ATTRIBUTE__PCI_BUS_ID = 28;
+const int CHPL_GPU_ATTRIBUTE__PCI_DEVICE_ID = 29;
+const int CHPL_GPU_ATTRIBUTE__MEMORY_CLOCK_RATE = 30;
+const int CHPL_GPU_ATTRIBUTE__GLOBAL_MEMORY_BUS_WIDTH = 31;
+const int CHPL_GPU_ATTRIBUTE__L2_CACHE_SIZE = 32;
+const int CHPL_GPU_ATTRIBUTE__MAX_THREADS_PER_MULTIPROCESSOR = 33;
+const int CHPL_GPU_ATTRIBUTE__COMPUTE_CAPABILITY_MAJOR = 34;
+const int CHPL_GPU_ATTRIBUTE__COMPUTE_CAPABILITY_MINOR = 35;
+const int CHPL_GPU_ATTRIBUTE__MAX_SHARED_MEMORY_PER_MULTIPROCESSOR = 36;
+const int CHPL_GPU_ATTRIBUTE__MANAGED_MEMORY = 37;
+const int CHPL_GPU_ATTRIBUTE__MULTI_GPU_BOARD = 38;
+const int CHPL_GPU_ATTRIBUTE__PAGEABLE_MEMORY_ACCESS = 39;
+const int CHPL_GPU_ATTRIBUTE__CONCURRENT_MANAGED_ACCESS = 40;
+const int CHPL_GPU_ATTRIBUTE__PAGEABLE_MEMORY_ACCESS_USES_HOST_PAGE_TABLES = 41;
+const int CHPL_GPU_ATTRIBUTE__DIRECT_MANAGED_MEM_ACCESS_FROM_HOST = 42;
+
+int chpl_gpu_impl_query_attribute(int dev, int attribute) {
+  chpl_internal_error(
```

Comment:
Should this be a warning instead to make CPU-as-device mode closer to actual GPU behavior?